src/field_theory/finite/trace.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2022 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import ring_theory.trace
+import field_theory.finite.basic
+import field_theory.finite.galois_field
+
+/-!
+# The trace map for finite fields
+
+We define `trace_to_zmod F` for a finite field `F` as the trace map
+from `F` to its prime field `zmod p` (where `p = ring_char F`),
+and we state the fact that this trace map is nondegenerate.
+
+## Tags
+finite field, trace
+-/
+
+namespace finite_field
+
+/-- The trace map from a finite field of characteristic `p` to `zmod p` -/
+noncomputable
+def trace_to_zmod (F : Type*) [field F] :
+ F →ₗ[zmod (ring_char F)] zmod (ring_char F) :=
+begin
+  letI := zmod.algebra F (ring_char F),
+  exact algebra.trace (zmod (ring_char F)) F,
+end
```

Comment:
```suggestion
algebra.trace (zmod (ring_char F)) F
```
This works for me, so I am wondering whether we need this specific definition at all.

Reply:
It does now for me, too. Possibly the necessary instance got added in the meantime.
I'll remove the definition and change the statement (and proof) of the lemma accordingly.